WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2538,6 +2540,23 @@ public void done() {
         }
     }
 
+    private void setFeaturedImageId(final long mediaId) {
+        mPost.setFeaturedImageId(mediaId);
+        if (mEditPostSettingsFragment != null) {
+            mEditPostSettingsFragment.updateFeaturedImage(mediaId);
+        } else {
+            // fragment may not exist if device was rotated while choosing featured image, so try again
+            // after a brief delay
+            new Handler().postDelayed(new Runnable() {
+                @Override public void run() {
+                    if (!isFinishing() && mEditPostSettingsFragment != null) {
+                        mEditPostSettingsFragment.updateFeaturedImage(mediaId);
+                    }
+                }
+            }, 250);
```

Comment:
I think this feel like a hack, it'd be better to just remember the `mediaId` value in a class member and then set it to `mEditPostSettingsFragment` as soon as such an instance becomes available. I've made a very brief experiment to try this out here, seems to be working fine https://github.com/wordpress-mobile/WordPress-Android/compare/issue/7623-stock-photo-rotation...issue/7623-stock-photo-rotation-mariotest?expand=1

Reply:
I admit to feeling I was creating a hack with that fix, but I'm also not wild about adding a "workaround variable" to this activity. I did a bit more testing, and the code below also appears to work and to me seems less hacky. WDYT?

```
private void setFeaturedImageId(final long mediaId) {
    mPost.setFeaturedImageId(mediaId);
    savePostAsync(new AfterSavePostListener() {
        @Override
        public void onPostSave() {
            EditPostActivity.this.runOnUiThread(new Runnable() {
                @Override public void run() {
                    if (mEditPostSettingsFragment != null) {
                        mEditPostSettingsFragment.updateFeaturedImage(mediaId);
                    }
                }
            });
        }
    });
}
```